rust/agama-server/src/questions/web.rs

```diff
@@ -0,0 +1,271 @@
+                .context("Failed to create interface name for question with password")?,
+        );
+        for (path, interfaces_hash) in objects.iter() {
+            if interfaces_hash.contains_key(&password_interface) {
+                result.push(self.create_question_with_password(path).await?)
+            } else {
+                result.push(self.create_generic_question(path).await?)
+            }
+        }
+        Ok(result)
+    }
+
+    async fn create_generic_question(
+        &self,
+        path: &OwnedObjectPath,
+    ) -> Result<Question, ServiceError> {
+        let dbus_question = GenericQuestionProxy::builder(&self.connection)
+            .path(path)?
+            .cache_properties(zbus::CacheProperties::No)
+            .build()
+            .await?;
+        let result = Question {
+            generic: GenericQuestion {
+                id: dbus_question.id().await?,
+                class: dbus_question.class().await?,
+                text: dbus_question.text().await?,
+                options: dbus_question.options().await?,
+                default_option: dbus_question.default_option().await?,
+                data: dbus_question.data().await?,
+            },
+            with_password: None,
+        };
+
+        Ok(result)
+    }
+
+    async fn create_question_with_password(
+        &self,
+        path: &OwnedObjectPath,
+    ) -> Result<Question, ServiceError> {
+        let dbus_question = QuestionWithPasswordProxy::builder(&self.connection)
+            .path(path)?
+            .cache_properties(zbus::CacheProperties::No)
+            .build()
+            .await?;
+        let mut result = self.create_generic_question(path).await?;
+        result.with_password = Some(QuestionWithPassword {
+            password: dbus_question.password().await?,
+        });
+
+        Ok(result)
+    }
+
+    pub async fn answer(&self, id: u32, answer: Answer) -> Result<(), ServiceError> {
+        let question_path = OwnedObjectPath::from(
+            ObjectPath::try_from(format!("/org/opensuse/Agama1/Questions/{}", id))
+                .context("Failed to create dbus path")?,
+        );
+        if let Some(password) = answer.with_password {
+            let dbus_password = QuestionWithPasswordProxy::builder(&self.connection)
+                .path(&question_path)?
+                .cache_properties(zbus::CacheProperties::No)
+                .build()
+                .await?;
+            dbus_password
+                .set_password(password.password.as_str())
+                .await?
+        }
+        let dbus_generic = GenericQuestionProxy::builder(&self.connection)
+            .path(&question_path)?
+            .cache_properties(zbus::CacheProperties::No)
+            .build()
+            .await?;
+        dbus_generic
+            .set_answer(answer.generic.answer.as_str())
+            .await?;
+        Ok(())
+    }
+}
+
+#[derive(Error, Debug)]
```

Comment:
This error do not bring any value. As we already did in the software layer, you can directly use [crate::error::Error](https://github.com/openSUSE/agama/blob/architecture_2024/rust/agama-server/src/error.rs#L12).